src/main/java/DBAccesser/User/UserAccessor.java

```diff
@@ -0,0 +1,106 @@
+        String SQLStatment = "SELECT Username FROM User WHERE Username=@Username OR EmailID=@EmailID";
+        Statement statement = Statement.newBuilder(SQLStatment)
+                                .bind("Username")
+                                .to(username)
+                                .bind("EmailID")
+                                .to(emailID)
+                                .build();
+        return !spannerTemplate
+                .query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true)) //setAllowPartialRead for reading specific columns
+                .isEmpty();
+    }
+
+    /* Checks if there is a row in the User table having the given UserID */
+    public boolean checkIfUserIDExists(long id) {
+        String SQLStatment = "SELECT UserID FROM User WHERE UserID=@userID";
+        Statement statement = Statement.newBuilder(SQLStatment)
+                                .bind("UserID")
+                                .to(id)
+                                .build();
+        return !spannerTemplate
+                .query(User.class, statement,  new SpannerQueryOptions().setAllowPartialRead(true))
+                .isEmpty();
+    }
+
+    /* Retrieves UserID of user having the given username */
+    public long getUserIDFromUsername(String username) {
+        String SQLStatment = "SELECT UserID FROM User WHERE Username=@Username";
+        Statement statement = Statement.newBuilder(SQLStatment)
+                                .bind("Username")
+                                .to(username)
+                                .build();
+        return spannerTemplate
+                .query(User.class, statement,  new SpannerQueryOptions().setAllowPartialRead(true))
+                .get(0).getUserID();
+    }
+
+    /* Retrieves the UserID of the user having the given username and password
+        If no such user exists, returns -1 */
+    public long login(String username, String password) {
+        String SQLStatment = "SELECT UserID from User WHERE Username=@username AND Password=@password";
+        Statement statement = Statement.newBuilder(SQLStatment)
+                                .bind("username")
+                                .to(username)
+                                .bind("password")
+                                .to(password)
+                                .build();
+        List<User> resultSet = spannerTemplate.query(User.class, statement,  new SpannerQueryOptions().setAllowPartialRead(true));
+        if(resultSet.isEmpty()) {
+            return -1;
+        }
+        return resultSet.get(0).getUserID();
+    }
+
+    /* Retrieves UserID, Username, EmailID, MobileNo and Picture of the user having the given username
+        If no such user exists, returns just the UserID as -1 */
+    public Map<String, Object> getUser(String username) {
+        String SQLStatment = "SELECT UserID, Username, EmailID, MobileNo, Picture FROM User WHERE Username=@Username";
```

Comment:
We can make this as static final variable within this class, as this is constant. Same for SQLStatement in other methods.

Reply:
I think we can not make a variable that is local to a method as static. And I am not sure how we can make `SQLStatment` as a class variable, because it's value is different for all methods like `getUser`, `login` etc.